### ISSUE TITLE
Error after update 0.21.5 in TimberImage

### ISSUE THREAD

**Issue by xroyx** [jarednova/timber#590]
Hi,

After updating to 0.21.5 I'm getting a bug in TimberImage 

    PHP Catchable fatal error: Method TimberImage::__toString() must return a string value in /sites/site/www/assets/plugins/timber-library/vendor/twig/twig/lib/Twig/Environment.php(332) : eval()'d code on line 40

Returning images with return value set to Image URL still works, but Image ID doesn't, for example:

    <img src="{{ TimberImage(block.image) }}">

Thanks,

**Comment by jarednova** [comment]
on it!

**Comment by jarednova** [comment]
@xroyx I created a test in 50feaf6 that recreates the scenario you outline, but I'm not getting the error. You said that `block.image` is returning an image ID? can you post some more about the particular image or how it's created so I can do some more looking?

**Comment by xroyx** [comment]
@jarednova The image is created with an ACF flexible content field:

Loading all the flexible content is done in page.twig with the following code:

    {% for block in post.get_field('blocks') %}
	    {% include 'views/content/' ~ block.acf_fc_layout ~ '.twig' %}
    {% endfor %}

And the image block is loaded with:

    <img src="{{ TimberImage(block.image) }}" alt="{{ TimberImage(block.image.alt) }}">

I figured out later I was using .alt in a wrong way.

    <img src="{{ TimberImage(block.image) }}" alt="{{ TimberImage(block.image).alt }}">

This was also the cause for the error and my themes not showing any flexible content. The wrong line of code didn't break my theme until the update to 0.21.5.

Sorry for the inconvenience and thank you for your time!

**Comment by jarednova** [comment]
I assume that `image` is an ACF Image field within your flexible content? Can you let me know...

- What version of ACF are you using?
- What is the Return Value of the image sub-field? The possibilities are: Image Array, Image URL or Image ID

... on the second point, if it works with one of the return values, but not another, that's very helpful to know

Thanks!

**Comment by xroyx** [comment]
@jarednova The image is indeed an ACF Image field. I'm using the latest version of ACF Pro (5.2.7). The return value is set to Image ID (so it can fetch alt data, etc.).

**Comment by jarednova** [comment]
great, thanks!

**Comment by tommymarshall** [comment]
Running into this too.

**Comment by tommymarshall** [comment]
I am having to check to make sure the `file` key exists before displaying an image to work around this: 
```twig
{% if post.hero_image %}
  {% set hero_image = TimberImage(post.hero_image) %}
  {% if hero_image.file %}
    <picture>
      <!--[if IE 9]><video style="display: none;"><![endif]-->
      <source srcset="{{ hero_image|resize(480, 320, 'center') }}" media="(min-width: 480px)">
      <source srcset="{{ hero_image|resize(600, 400, 'center') }}" media="(min-width: 600px)">
      <!--[if IE 9]></video><![endif]-->
      <img alt="" srcset="{{ hero_image|resize(320, 213, 'center') }}">
    </picture>
  {% endif %}
{% endif %}
```

Without the extra check on line 3, it throws the toString fatal error. It'd be nice if this simply returned an empty object and supressed the error.

**Comment by jarednova** [comment]
@xroyx I'm still not able to re-create the issue but I've got a pretty good feeling that this fix will solve the issue for you:

https://github.com/jarednova/timber/pull/599

... can you give it a shot and see if that resolves the error?

**Comment by jarednova** [comment]
@xroyx @tommymarshall YES! I was able to replicate the issue and fix. The PR is #599. After it clears the Travis code checks I will merge into master

**Comment by tommymarshall** [comment]
Awesome! Thanks @jarednova

**Comment by xroyx** [comment]
@jarednova Will check it out after the weekend when I have access to my development server again. Thanks a lot!

**Comment by jarednova** [comment]
Closed by #599
